subgraphs/burrow/src/helpers/rates.ts

```diff
@@ -1,26 +1,49 @@
-import { near } from '@graphprotocol/graph-ts';
-import { Market, InterestRate } from '../../generated/schema';
+import { near } from "@graphprotocol/graph-ts";
+import { Market, InterestRate } from "../../generated/schema";
+import {
+  InterestRateSide,
+  InterestRateType,
+  NANOS_TO_HOUR,
+} from "../utils/const";
 
-export function getOrCreateSupplyRate(market: Market, receipt: near.ReceiptWithOutcome|null = null): InterestRate {
-	let id = 'SUPPLY-VARIABLE-'.concat(market.id);
-	if(receipt) id = id.concat('-').concat((receipt.block.header.timestampNanosec / 86400000000000).toString());
-	let rate = InterestRate.load(id);
-	if (rate == null) {
-		rate = new InterestRate(id);
-		rate.side = 'LENDER';
-		rate.type = 'VARIABLE';
-	}
-	return rate as InterestRate;
+export function getOrCreateSupplyRate(
```

Comment:
Why not make this and the function below all one function?

Reply:
Can you make it one function, essentially you pass in the InterestRateSide as a variable.

That way it is easier to maintain in the long term. Imagine we change the interest rate sides or add more you would need to make a new function for each one